src/frontend/src/lib/components/registry/players/PlayerTournamentHistory.svelte

```diff
@@ -0,0 +1,228 @@
+  let game: string | null = null;
+  let from: string | null = null;
+  let to: string | null = null;
+  let solo_placements: PlayerTournamentPlacement[] = [];
+  let team_placements: PlayerTournamentPlacement[] = [];
+  let filtered_solo_placements: PlayerTournamentPlacement[] = [];
+  let filtered_team_placements: PlayerTournamentPlacement[] = [];
+  // Default 'silver' from PlacementsDisplay.svelte is less readable than I'd like
+  let podium_style: { [key: number]: string } = { 1: 'gold', 2: 'bg-slate-400/60', 3: 'bronze' };
+
+  let avatar_url = logo;
+  if (player.user_settings && player.user_settings.avatar) {
+    avatar_url = player.user_settings.avatar;
+  }
+
+  function toDate(unix_timestamp: number) {
+    return new Date(unix_timestamp * 1000).toLocaleDateString();
+  }
+
+  async function fetchData() {
+    // API
+    let url = `/api/tournaments/players/placements/${player.id}`;
+    const res = await fetch(url);
+    if (res.status !== 200) {
+      return;
+    }
+    let body = await res.json();
+    team_placements = body.tournament_team_placements;
+    solo_placements = body.tournament_solo_and_squad_placements;
+
+    // Filtering
+    filtered_team_placements = [...team_placements];
+    filtered_solo_placements = [...solo_placements];
+
+    if (game) {
+      filtered_team_placements = filtered_team_placements.filter((item) => item.game === game);
+      filtered_solo_placements = filtered_solo_placements.filter((item) => item.game === game);
+    }
+    if (mode) {
+      filtered_team_placements = filtered_team_placements.filter((item) => item.mode === mode);
+      filtered_solo_placements = filtered_solo_placements.filter((item) => item.mode === mode);
+    }
+    if (from) {
+      filtered_solo_placements = filtered_solo_placements.filter((item) => {
+        return item.date_start >= Date.parse(from) / 1000;
+      });
+    }
+    if (to) {
+      filtered_team_placements = filtered_team_placements.filter((item) => {
+        return item.date_end <= Date.parse(to) / 1000;
+      });
+    }
+
+    // Sorting
+    filtered_solo_placements = filtered_solo_placements.sort((a, b) => {
+      return b.date_start - a.date_start;
+    });
+
+    filtered_team_placements = filtered_team_placements.sort((a, b) => {
+      return b.date_start - a.date_start;
+    });
+  }
+
+  onMount(fetchData);
+</script>
+
+<Section header={$LL.TOURNAMENT_HISTORY.TOURNAMENT_HISTORY()}>
+  <div class="w-full m-auto">
+    <form on:submit|preventDefault={fetchData}>
+      <div class="flex flex-row flex-wrap items-center justify-center">
+        <GameModeSelect bind:game bind:mode all_option hide_labels is_team />
+        <div class="flex flex-col">
+          <div class="ml-1">
+            <input class="w-44" name="from" type="date" bind:value={from} />
+          </div>
+          <div class="ml-1">
+            <input class="w-44" name="to" type="date" bind:value={to} />
+          </div>
+        </div>
```

Comment:
From and To inputs should have labels for clarity imo, also should try to have all of the selects/inputs fit on one line with flex instead of the 2-row layout it has currently:
![image](https://github.com/user-attachments/assets/57ee1566-fcbe-4f20-a0bc-9c5892e2fe2a)

Reply:
I disagree. The design of the site as it is now has everything flowing from top to bottom. I think it's reasonable for a user to expect a date range to follow the same pattern. I settled on this implementation due to trying to keep the styling as consistent as possible while leveraging the top-to-bottom patterns of the rest of the site. 

My thoughts:
- Game & Mode pickers are labels in themselves, and the `<input type="date">` html element doesn't seem to support placeholders of different types (otherwise I would put a label inside the picker itself for consistent styling - problem solved)
- Labels on top of the pickers make the other pickers misaligned vertically. Looks bad imo.
- Labels before (on the left of) the pickers make the flex containers wrap in undesirable ways at certain sizes, for example:
```
All Games | All Modes | From Date
To Date
```
and I would guess the majority of users are mobile, so designing with them in mind is important to me. The above example also applies to having the pickers in a straight line.
- Labels after (on the right of) the pickers doesn't make sense.

___

If you have a UI idea in mind please let me know! I could not come up with a solution that I liked aside from this one. 🤷

Reply:
I tried something like this as a concept:
![image](https://github.com/user-attachments/assets/1235f451-6c52-4199-91f0-1b1b914b3ca0)
![image](https://github.com/user-attachments/assets/842083e0-f373-40d9-8232-f48672fe9393)

Admittedly I am not very good at CSS so it doesn't look great on mobile but you kinda get the idea. I tried getting it in one line for PC width but ran into some issues with the flex containers like you said so I compromised with keeping your 2-line approach. If you can figure out a way to get everything to look aligned with a similar layout to this I think it could be good, otherwise I think you make good points so you could just leave it be if you want. Here's the code I tried:
```python
        <GameModeSelect bind:game bind:mode all_option hide_labels is_team />
        <div class="flex flex-col ml-1">
          <div class="ml-1">
            <label for="from" class="inline-block w-12">From</label>
            <input class="w-52" name="from" type="date" bind:value={from} />
          </div>
          <div class="ml-1">
            <label for="to" class="inline-block w-12">To</label>
            <input class="w-52" name="to" type="date" bind:value={to} />
          </div>
        </div>```